warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/SplitsFile.java

```diff
@@ -0,0 +1,184 @@
+package datawave.ingest.mapreduce.job;
+
+import datawave.ingest.mapreduce.handler.shard.ShardIdFactory;
+import datawave.ingest.mapreduce.handler.shard.ShardedDataTypeHandler;
+import datawave.util.time.DateHelper;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.commons.lang.time.DateUtils;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SplitsFile {
+    private static final Logger log = Logger.getLogger(SplitsFile.class);
+    
+    public static final String SPLIT_WORK_DIR = "split.work.dir";
+    public static final String MAX_SHARDS_PER_TSERVER = "shardedMap.max.shards.per.tserver";
+    public static final String SHARD_VALIDATION_ENABLED = "shardedMap.validation.enabled";
+    public static final String SHARDS_BALANCED_DAYS_TO_VERIFY = "shards.balanced.days.to.verify";
+    public static final String CONFIGURED_SHARDED_TABLE_NAMES = ShardedDataTypeHandler.SHARDED_TNAMES + ".configured";
+    
+    public static void setupFile(Configuration conf) throws IOException, URISyntaxException, AccumuloSecurityException, AccumuloException {
+        
+        Path baseSplitsPath = TableSplitsCache.getSplitsPath(conf);
+        FileSystem sourceFs = baseSplitsPath.getFileSystem(conf);
+        FileSystem destFs = new Path(conf.get(SPLIT_WORK_DIR)).getFileSystem(conf);
+        
+        // want validation turned off by default
+        boolean doValidation = conf.getBoolean(SHARD_VALIDATION_ENABLED, false);
+        
+        try {
+            log.info("Base splits: " + baseSplitsPath);
+            
+            Path destSplits = new Path(conf.get(SPLIT_WORK_DIR) + "/"
+                            + conf.get(TableSplitsCache.SPLITS_CACHE_FILE, TableSplitsCache.DEFAULT_SPLITS_CACHE_FILE));
+            log.info("Dest splits: " + destSplits);
+            
+            FileUtil.copy(sourceFs, baseSplitsPath, destFs, destSplits, false, conf);
```

Comment:
This is preserving the current workflow.  setupFile is called by IngestJob, where it uses or updates the base splits file, then generates copies in the work dir.  The mappers reference the splits files in their job's work dir to avoid a thundering herd against the base file only.  I opted to keep this behavior the same for now, but we could change it and use distributedcache or some other mechanism.